schema/rpc.fbs

```diff
@@ -521,3 +521,22 @@ table SetPauseTrackingRequest {
 
 /// Clears mounting reset data, defaulting to the manually set mounting orientations
 table ClearMountingResetRequest {}
+
+/// Common folders often used in computers for storing files
+enum ComputerDirectory: uint8 {
+    DOCUMENTS,
+}
+
+/// Used for the server to save a file and have it prompt in the user side
+table SaveFileNotification {
+    /// Binary data of the file
+    data: [uint8] (required);
```

Comment:
dont make fields required, its an antipattern

Reply:
It's not in this case, why would it be an anti pattern of a file save to contain a blob. When would it be null?

Reply:
Because there is no downside to it being non-required for starters. The field is already behind an indirection, its not saving anything to make it required afaik. And lets say we realize that actually maybe we should pass a URI to a file to save instead? You make a URI field, which is fine, but wait we can't indicate that the `data` field should not be present. Instead we set it to a zero length vector, and add special logic in to handle the case.

In general, there is absolutely zero scenario where marking a field as required would be both more space efficient and also less brittle. This is admittedly an overly opinionated stance, but its a stance I'm taking here.

You can read more about this debate here if you want to go down the internet flame war hole: https://stackoverflow.com/questions/31801257/why-required-and-optional-is-removed-in-protocol-buffers-3

Reply:
We can't pass an URI, we need to still support web GUI...
But ok, I will support null + zero length vector because maintaining it is not painful at all >~>

Reply:
fixed